qcodes/instrument/ip.py

```diff
@@ -1,47 +1,51 @@
 """Ethernet instrument driver class based on sockets."""
 import socket
 import logging
-from typing import Dict, Sequence, Optional
+from typing import Dict, Sequence, Optional, Union, Any
 
 from .base import Instrument
 
 log = logging.getLogger(__name__)
 
 
 class IPInstrument(Instrument):
-
     r"""
     Bare socket ethernet instrument implementation.
 
     Args:
-        name (str): What this instrument is called locally.
+        name: What this instrument is called locally.
 
-        address (Optional[str]): The IP address or name. If not given on
+        address: The IP address or name. If not given on
             construction, must be provided before any communication.
 
-        port (Optional[int]): The IP port. If not given on construction, must
+        port: The IP port. If not given on construction, must
             be provided before any communication.
 
-        timeout (number): Seconds to allow for responses. Default 5.
+        timeout: Seconds to allow for responses. Default 5.
 
-        terminator (str): Character(s) to terminate each send. Default '\n'.
+        terminator: Character(s) to terminate each send. Default '\n'.
 
-        persistent (bool): Whether to leave the socket open between calls.
+        persistent: Whether to leave the socket open between calls.
             Default True.
 
-        write_confirmation (bool): Whether the instrument acknowledges writes
+        write_confirmation: Whether the instrument acknowledges writes
             with some response we should read. Default True.
 
-        metadata (Optional[Dict]): additional static metadata to add to this
+        kwargs: additional static metadata to add to this
             instrument's JSON snapshot.
 
     See help for ``qcodes.Instrument`` for additional information on writing
     instrument subclasses.
     """
 
-    def __init__(self, name, address=None, port=None, timeout=5,
-                 terminator='\n', persistent=True, write_confirmation=True,
-                 **kwargs):
+    def __init__(self, name: str,
+                 address: Optional[str] = None,
+                 port: Optional[int] = None,
+                 timeout: Union[float, int] = 5,
```

Comment:
Should we have an alias for that here too

Reply:
I think this is not necessary if I understand this correctly:
https://www.python.org/dev/peps/pep-0484/#the-numeric-tower
so it should simply be float.
There are some more ocurences and this would imply we can also remove the home defined `Number`  type.